File: core-common/src/main/java/com/huyingbao/core/base/view/BaseDialog.java
package com.huyingbao.core.base.view;

import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.Window;

import com.huyingbao.core.arch.view.RxFluxDialog;
import com.huyingbao.core.base.BaseView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModel;
import butterknife.ButterKnife;
import butterknife.Unbinder;

/**
 * Created by liujunfeng on 2019/1/1.
 */
public abstract class BaseDialog<T extends ViewModel> extends RxFluxDialog<T> implements BaseView {
    private Unbinder mUnbinder;

    @NonNull
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        //注意此处android.R.id.content
        return inflater.inflate(getLayoutId(), getDialog().getWindow().findViewById(android.R.id.content), false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        //ButterKnife绑定
        mUnbinder = ButterKnife.bind(this, view);
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        // view创建之后的操作
        afterCreate(savedInstanceState);
    }

    @Override
    public void onStart() {
        super.onStart();
        final Window window = getDialog().getWindow();
        //设置Dialog对应的布局背景和尺寸生效
        window.setBackgroundDrawable(new ColorDrawable(Color.TRANSPARENT));
        window.setLayout(ViewGroup.LayoutParams.WRAP_CONTENT, ViewGroup.LayoutParams.WRAP_CONTENT);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mUnbinder.unbind();
    }
}

File: core-base/src/main/java/com/huyingbao/core/base/BaseDialog.kt
package com.huyingbao.core.base

import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatDialogFragment
import com.huyingbao.core.base.BaseView

/**
 * 不使用Dagger.Android，不持有ViewModel，不自动管理订阅
 *
 * 布局文件值与实际显示一致
 *
 * Created by liujunfeng on 2019/1/1.
 */
abstract class BaseDialog() :
    AppCompatDialogFragment(),
    BaseView {
    /**
     * 注意此处android.R.id.content，配合[BaseDialog.onStart]方法，使布局文件中背景值和尺寸值生效
     */
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return inflater.inflate(
            getLayoutId(),
            dialog?.window?.findViewById(android.R.id.content),
            false
        )
    }

    /**
     * DialogFragment内存泄露解决方案
     */
    override fun onActivityCreated(savedInstanceState: Bundle?) {
        if (showsDialog) {
            showsDialog = false
        }
        super.onActivityCreated(savedInstanceState)
        showsDialog = true

        val view = view
        if (view != null) {
            if (view.parent != null) {
                throw IllegalStateException("DialogFragment can not be attached to a container view")
            }
            dialog!!.setContentView(view)
        }
        val activity = activity
        if (activity != null) {
            dialog!!.setOwnerActivity(activity)
        }
        if (savedInstanceState != null) {
            val dialogState = savedInstanceState.getBundle("android:savedDialogState")
            if (dialogState != null) {
                dialog!!.onRestoreInstanceState(dialogState)
            }
        }
        afterCreate(savedInstanceState)
    }

    /**
     * Dialog对应的布局文件中背景值和尺寸值生效
     */
    override fun onStart() {
        dialog?.window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
        dialog?.window?.setLayout(
            ViewGroup.LayoutParams.WRAP_CONTENT,
            ViewGroup.LayoutParams.WRAP_CONTENT
        )
        super.onStart()
    }
}